cpu/nrf52/periph/i2c.c

```diff
@@ -167,29 +173,36 @@ void i2c_release(i2c_t dev)
 int i2c_write_regs(i2c_t dev, uint16_t addr, uint16_t reg,
                    const void *data, size_t len, uint8_t flags)
 {
-    assert((dev < I2C_NUMOF) && data && (len > 0) && (len < 255));
+    assert((dev < I2C_NUMOF) && data && (len > 0) && (len < 253));
 
-    if (flags & (I2C_NOSTART | I2C_REG16 | I2C_ADDR10)) {
+    if (flags & (I2C_NOSTART | I2C_ADDR10)) {
         return -EOPNOTSUPP;
     }
-    /* the nrf52's TWI device does not support to do two consecutive transfers
-     * without a repeated start condition in between. So we have to put all data
-     * to be transferred into a temporary buffer
-     *
-     * CAUTION: this might become critical when transferring large blocks of
-     *          data as the temporary buffer is allocated on the stack... */
```

Comment:
We should retain this comment so people know why this buffer is there in the first place.

Reply:
I think it should be modified then.

There's no temporary buffer allocated, the buffer is permanent and has a fixed size whether or not we transfer large blocks.